File: Data Structures & Algorithms/jump-game/submission-0.cs
public class Solution {
    public bool CanJump(int[] nums) {
        var target = nums.Length - 1;

        bool dfs(int ind)
        {
            if (ind == target) return true;
            if (nums[ind] == 0) return false;
            if (ind >= nums.Length) return false;


            for(var i = 1; i <= nums[ind]; i++)
            {
                if (dfs(ind + i)) return true;
            }

            return false;
        }

        return dfs(0);

    }
}

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        
        def binS(s: int, e: int):
            if s == e and nums[s] != target:
                return -1

            im = (e - s) // 2 + s
            m = nums[im]
            if m == target:
                return im
            
            if m > target:
                return binS(s, im)
            else:
                return binS(im + 1, e)
        
        return binS(0, len(nums) - 1)
